whois-nrtm/src/main/java/net/ripe/db/whois/nrtm/NrtmQueryHandler.java

```diff
@@ -260,11 +262,15 @@ private void handleVersionQuery(final Channel channel) {
 
     @Override
     public void channelActive(final ChannelHandlerContext ctx) throws Exception {
-        PendingWrites.add(ctx.channel());
+        if (!ctx.channel().hasAttr(TERMS_CONDITIONS)) {
+            PendingWrites.add(ctx.channel());
+
+            writeMessage(ctx.channel(),  NrtmMessages.termsAndConditions());
 
-        writeMessage(ctx.channel(),  NrtmMessages.termsAndConditions());
+            ctx.channel().attr(TERMS_CONDITIONS).set(true);
```

Comment:
Is this attribute ever read?

Reply:
Yes in the channel active for t&c, same reasons as the T&C handler.